### PR TITLE
Fix bunch of issues

### PR DESCRIPTION
**Pull request by DecampsRenan**
## Describe your changes

- 🐛 Fix missing `tsx` dev dependency which causes the `yarn db:seed` to fail
- 🐛 Fix bad check on `NEXT_PUBLIC_IS_DEMO` env var, which causes the `DemoRegisterHint` to always show the demo mode alert
- 🐛 Fix bad check on `key` search param on activate user route, which causes the activation to always fail

## Checklist

 - [x] I performed a self review of my code
 - [x] I ensured that everything is written in English
 - [x] I tested the feature or fix on my local environment
 - [x] I ran the `yarn storybook` command and everything is working

### CODE REVIEW

**Comment by yoannfleurydev** [comment]
Are you sure about that one? `npx` should propose to download the binary if it is not present.

image

**Comment by DecampsRenan** [comment]
> Are you sure about that one? `npx` should propose to download the binary if it is not present.
> 
> image

Yes that's my understanding of npx to ^^ but it seems it doesn't :/ maybe it's an issue on my side and not related in start ui

For the record:
image

File: src/features/demo-mode/DemoLoginHint.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const DemoLoginHint = () => {
   const mockedUsername = 'admin';
   const mockedPassword = 'admin';
 
-  if (!process.env.NEXT_PUBLIC_IS_DEMO) return null;
+  if (process.env.NEXT_PUBLIC_IS_DEMO !== 'true') return null;
```
Comment (DecampsRenan) [inline src/features/demo-mode/DemoLoginHint.tsx:15]: I think it's a hotfix and we can do better (maybe by transforming it to a real boolean or at least doing a getter)

**Comment by DecampsRenan** [comment]
> > Are you sure about that one? `npx` should propose to download the binary if it is not present.
> > image
> 
> Yes that's my understanding of npx to ^^ but it seems it doesn't :/ maybe it's an issue on my side and not related in start ui
> 
> For the record: image

I did some checks (using zsh instead of fish, trying the npx command directly)
It seems it fails when the command is run from yarn; adding it as a dependency fixed the issue :/

It could be nice if someone else could try this part of the setup, just to be sure it's not on my side

File: src/features/demo-mode/DemoLoginHint.tsx (hunk 1 above)
Comment (yoannfleurydev) [reply src/features/demo-mode/DemoLoginHint.tsx:15]: strings are ok for env var to be honest, looks good to me like that

**Comment by yoannfleurydev** [comment]
> > > Are you sure about that one? `npx` should propose to download the binary if it is not present.
> > > image
> > 
> > 
> > Yes that's my understanding of npx to ^^ but it seems it doesn't :/ maybe it's an issue on my side and not related in start ui
> > For the record: image
> 
> I did some checks (using zsh instead of fish, trying the npx command directly) It seems it fails when the command is run from yarn; adding it as a dependency fixed the issue :/
> 
> It could be nice if someone else could try this part of the setup, just to be sure it's not on my side

Thanks for your work, I'm gonna merge it like that to unblock potential problems for users. We'll improve that part later if possible. It is a devDependency so it doesn't bother too much.

File: src/app/api/jhipster-mocks/activate/route.ts [hunk 2]
```diff
@@ -9,16 +9,14 @@ import {
 
 export const GET = apiMethod({
   public: true,
-  handler: async ({ params }) => {
-    const safeParams = z.object({ key: z.string() }).safeParse(params);
```
Comment (yoannfleurydev) [inline src/app/api/jhipster-mocks/activate/route.ts:13]: We can remove the `z` import as we are not using it anymore.

**Review by yoannfleurydev — APPROVED** [review]

**Pull request merged**